mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -3874,6 +3874,39 @@ fn test_electrum_display_balances() {
     block_on(utxo_common_tests::test_electrum_display_balances(&rpc_client));
 }
 
+#[test]
+fn test_for_non_existent_tx_hex_utxo_electrum() {
+    let client = electrum_client_for_test(RICK_ELECTRUM_ADDRS);
+    let coin = utxo_coin_for_test(
+        client.into(),
+        Some("spice describe gravity federal blast come thank unfair canal monkey style afraid"),
+        false,
+    );
+    let tx = vec![
+        4, 0, 0, 128, 133, 32, 47, 137, 1, 81, 146, 94, 240, 227, 116, 153, 194, 232, 25, 255, 192, 110, 155, 208, 14,
+        157, 197, 189, 65, 107, 229, 129, 124, 136, 215, 10, 63, 13, 218, 183, 206, 1, 0, 0, 0, 107, 72, 48, 69, 2, 33,
+        0, 255, 148, 80, 199, 101, 182, 102, 18, 1, 46, 26, 48, 212, 144, 162, 81, 21, 157, 233, 80, 159, 130, 184, 56,
+        98, 183, 189, 115, 233, 233, 30, 60, 2, 32, 124, 220, 236, 219, 101, 75, 139, 241, 185, 174, 16, 112, 127, 128,
+        159, 9, 104, 156, 39, 2, 81, 187, 198, 209, 255, 79, 84, 61, 238, 53, 243, 35, 1, 33, 3, 192, 45, 65, 142, 227,
+        32, 245, 206, 82, 10, 140, 237, 131, 171, 113, 63, 246, 56, 118, 88, 57, 29, 11, 37, 227, 59, 106, 136, 90,
+        252, 3, 72, 255, 255, 255, 255, 3, 0, 225, 245, 5, 0, 0, 0, 0, 23, 169, 20, 27, 130, 67, 246, 162, 41, 162,
+        144, 64, 214, 12, 227, 165, 136, 151, 82, 145, 176, 74, 245, 135, 0, 0, 0, 0, 0, 0, 0, 0, 22, 106, 20, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 24, 3, 129, 66, 23, 0, 0, 0, 25, 118, 169, 20, 158, 220,
+        162, 160, 79, 133, 71, 112, 148, 22, 186, 113, 239, 163, 18, 241, 138, 161, 220, 132, 136, 172, 174, 23, 247,
+        98, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    ]; // Bad transaction hex
```

Comment:
Does `bad` mean that the transaction cannot be deserialized?
Or does it mean that the transaction has a transaction hash that is not on the chain?
If the second option, then could you please use refactor it similar to this:
```rust
let tx_str = "0400008085202f8902bf17bf7d1daace52e08f732a6b8771743ca4b1cb765a187e72fd091a0aabfd52000000006a47304402203eaaa3c4da101240f80f9c5e9de716a22b1ec6d66080de6a0cca32011cd77223022040d9082b6242d6acf9a1a8e658779e1c655d708379862f235e8ba7b8ca4e69c6012102031d4256c4bc9f99ac88bf3dba21773132281f65f9bf23a59928bce08961e2f3ffffffffff023ca13c0e9e085dd13f481f193e8a3e8fd609020936e98b5587342d994f4d020000006b483045022100c0ba56adb8de923975052312467347d83238bd8d480ce66e8b709a7997373994022048507bcac921fdb2302fa5224ce86e41b7efc1a2e20ae63aa738dfa99b7be826012102031d4256c4bc9f99ac88bf3dba21773132281f65f9bf23a59928bce08961e2f3ffffffff0300e1f5050000000017a9141ee6d4c38a3c078eab87ad1a5e4b00f21259b10d870000000000000000166a1400000000000000000000000000000000000000001b94d736000000001976a91405aab5342166f8594baf17a7d9bef5d56744332788ac2d08e35e000000000000000000000000000000";
let tx: UtxoTx = tx_str.into();
```

Reply:
yes, correct for the later